Libraries/Network/XMLHttpRequest.js

```diff
@@ -334,21 +370,24 @@ class XMLHttpRequest extends EventTarget(...XHR_EVENTS) {
     if (!url) {
       throw new Error('Cannot load an empty url');
     }
-    this._reset();
     this._method = method.toUpperCase();
     this._url = url;
     this._aborted = false;
     this.setReadyState(this.OPENED);
   }
 
-  sendImpl(
-    method: ?string,
-    url: ?string,
-    headers: Object,
-    data: any,
-    useIncrementalUpdates: boolean,
-    timeout: number,
-  ): void {
+  send(data: any): void {
+    if (this.readyState !== this.OPENED) {
+      throw new Error('Request has not been opened');
+    }
+    if (this._sent) {
+      throw new Error('Request has already been sent');
+    }
+    this._sent = true;
+    const incrementalEvents = this._incrementalEvents ||
```

Comment:
We can't. `this._incrementalEvents` means a `readstatechange` or `progress` handler was set via `addEventListener()`. But it doesn't capture if either handler was set by setting the `onreadystatechange` or `onprogress` properties.